addons/knowledge/models/knowledge_article.py

```diff
@@ -800,20 +803,21 @@ def _invite_members(self, partner_ids, permission, send_mail=True):
                 self._send_invite_mail(partner)
 
     def _send_invite_mail(self, partner):
-        self.ensure_one()
-        subject = _("Invitation to access %s", self.name)
-        partner_lang = get_lang(self.env, lang_code=partner.lang).code
-        body = self.env['ir.qweb'].with_context(lang=partner_lang)._render('knowledge.knowledge_article_mail_invite', {
-            'record': self,
-            'user': self.env.user,
-            'recipient': partner,
-            'link': self._get_invite_url(partner),
-        })
+        # TODO: only send one email if user is invited to multiple articles?
+        for article in self:
```

Comment:
this method could be multi indeed, in case we need it later.
but for now, we still send one mail. As we don't need to send a mail to inform the partner he has been added as member with no access on the children articles (that the user who invited the partner has no clue they exist).
When you invite a partner, only one article is important, the one that have been share, not its potential children.